### PR TITLE
Retain attribute on pexp_setinstvar

### PR DESCRIPTION
**Pull request by CraigFe**
This attempts to resolve https://github.com/ocaml-ppx/ocamlformat/issues/1258.

It's unecessary to wrap `(a <- b) [@a]` in parentheses if the expression is the only one in a module definition.

I discovered and tested a few unnecessary parentheses in the process:

- `method a = (1 [@attr])`
- `method a = (0 [@attr]) ; 1`
- `let _ = (0 [@attr]) ; 1` (edit: now tested by https://github.com/ocaml-ppx/ocamlformat/pull/1291 instead)

We may want to keep these for disambiguation purposes, but it seems better to have them tested regardless.

### CODE REVIEW

**Comment by CraigFe** [comment]
The unnecessary parentheses mentioned here are removed as a side-effect of https://github.com/ocaml-ppx/ocamlformat/pull/1291.

File: lib/Ast.ml [hunk 1]
```diff
@@ -2279,6 +2279,7 @@ end = struct
           ; _ }
       , _ ) ->
         false
+    | Exp {pexp_desc= Pexp_object _; _}, _ -> false
```
Comment (Julow) [inline lib/Ast.ml]: Could you add a small comment explaining why we have this pattern ?

File: test/passing/object.ml [hunk 2]
```diff
@@ -6,6 +6,8 @@ let _ =
     (* some comment *)
     method! x = 2 [@@attr]
 
+    method x = (1 [@attr])
```
Comment (Julow) [inline test/passing/object.ml]: These parentheses are not necessary.

Comment (Julow) [reply test/passing/object.ml]: Is this fixed by https://github.com/ocaml-ppx/ocamlformat/pull/1291 ?

Comment (CraigFe) [reply test/passing/object.ml]: It does not, only those on the LHS of sequences will be, i.e.

```ocaml
method a = (0 [@attr]) ; 1

let _ = (0 [@attr]) ; 1
```

File: lib/Ast.ml (hunk 1 above)
Comment (CraigFe) [reply lib/Ast.ml]: Done in https://github.com/ocaml-ppx/ocamlformat/pull/1288/commits/ed75834457eb13065794a4806fff5074e36218a9.

**Comment by Julow** [comment]
Let's merge !

**Pull request merged**